lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -0,0 +1,138 @@
+
+        # Create a course and configure it as a credit course
+        self.course = CourseFactory.create()
+        CreditCourse.objects.create(course_key=self.course.id, enabled=True)  # pylint: disable=no-member
+
+        # Configure credit requirements (passing grade and in-course reverification)
+        credit_api.set_credit_requirements(
+            self.course.id,  # pylint: disable=no-member
+            [
+                {
+                    "namespace": "grade",
+                    "name": "grade",
+                    "display_name": self.MIN_GRADE_REQ_DISPLAY,
+                    "criteria": {
+                        "min_grade": 0.8
+                    }
+                },
+                {
+                    "namespace": "reverification",
+                    "name": "midterm",
+                    "display_name": self.VERIFICATION_REQ_DISPLAY,
+                    "criteria": {}
+                }
+            ]
+        )
+
+        # Create a user and log in
+        self.user = UserFactory.create(username=self.USERNAME, password=self.PASSWORD)
+        self.user.profile.name = self.USER_FULL_NAME
+        self.user.profile.save()
+
+        result = self.client.login(username=self.USERNAME, password=self.PASSWORD)
+        self.assertTrue(result, msg="Could not log in")
+
+        # Enroll the user in the course as "verified"
+        self.enrollment = CourseEnrollmentFactory(
+            user=self.user,
+            course_id=self.course.id,  # pylint: disable=no-member
+            mode="verified"
+        )
+
+    def test_credit_requirements_maybe_eligible(self):
+        # The user hasn't satisfied any of the credit requirements yet, but she
+        # also hasn't failed any.
+        response = self._get_progress_page()
+
+        # Expect that the requirements are displayed
+        self.assertContains(response, self.MIN_GRADE_REQ_DISPLAY)
+        self.assertContains(response, self.VERIFICATION_REQ_DISPLAY)
+        self.assertContains(response, "Upcoming")
+        self.assertContains(response, "Bob, you have not yet met the requirements for credit")
+
+    def test_credit_requirements_eligible(self):
+        # Mark the user as eligible for all requirements
+        credit_api.set_credit_requirement_status(
+            self.user.username, self.course.id,
+            "grade", "grade",
+            status="satisfied",
+            reason={"final_grade": 0.95}
+        )
+
+        credit_api.set_credit_requirement_status(
+            self.user.username, self.course.id,
+            "reverification", "midterm",
+            status="satisfied", reason={}
+        )
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        self.assertContains(response, self.MIN_GRADE_REQ_DISPLAY)
+        self.assertContains(response, self.VERIFICATION_REQ_DISPLAY)
+        self.assertContains(response, "Bob, you have met the requirements for credit in this course.")
```

Comment:
Same as above, I'd also turn this into a format string in order to make use of `USER_FULL_NAME`.